shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php

```diff
@@ -80,4 +80,48 @@ public function patchEntity($entity_id) {
     }
     return $entity;
   }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function checkEntityAccess($op, $entity_type, $entity) {
+    $account = $this->getAccount();
+
+    $wrapper = entity_metadata_wrapper('node', $entity);
+    $repo_name = $wrapper->og_repo->label();
+
+    if (og_is_member('node', $wrapper->og_repo->getIdentifier(), 'user', $account)) {
+      // User is member of the repository in shoov - he has access to build even
+      // if he doesn't have it in GitHub.
+     return TRUE;
+    }
+
+    // Check user is member of the repository in GitHub.
+    $user_wrapper = entity_metadata_wrapper('user', $account);
+    $user_name = $user_wrapper->label();
+    $access_token = $user_wrapper->field_github_access_token->value();
+
+    $options = array(
+      'method' => 'GET',
+      'headers' => array(
+        'Authorization' => 'token ' . $access_token,
+      ),
+    );
+    $url = 'repos/' . $repo_name . '/collaborators/' . $user_name;
+    $response = shoov_github_http_request($url, $options);
+
+    if ($response['meta']['status'] == 204) {
+      // User is a member of the repository. Subscribe him to the repository.
+      $params = array(
+        'entity_type' => 'user',
+        'entity' => $account,
+        'field_name' => 'og_user_node'
+      );
+
+      og_group('node', $wrapper->og_repo->getIdentifier(), $params);
+      return TRUE;
+    }
+
+    return FALSE;
```

Comment:
But if someone wants user to have access to the build even if user doesn't have access in github?